### PR TITLE
Reduced docker image size

### PR DESCRIPTION
**Pull request by taras-keryk-plv**
Updated dockerfiles to reduce image size

### CODE REVIEW

File: Dockerfile.saithrift [hunk 1]
```diff
@@ -15,43 +11,14 @@ ENV DEBIAN_FRONTEND=noninteractive
...
-        git \
-        procps \
-        build-essential \
-        graphviz \
-        doxygen \
-        aspell \
-        python3-pip \
-        rsyslog \
-        supervisor \
         cmake \
```
Comment (andriy-kokhan) [inline Dockerfile.saithrift:14]: in base image

File: Dockerfile.saithrift [hunk 2]
```diff
@@ -15,43 +11,14 @@ ENV DEBIAN_FRONTEND=noninteractive
...
-        graphviz \
-        doxygen \
-        aspell \
         git \
```
Comment (andriy-kokhan) [inline Dockerfile.saithrift:15]: in base image

File: Dockerfile.saithrift [hunk 3]
```diff
@@ -15,43 +11,14 @@ ENV DEBIAN_FRONTEND=noninteractive
...
-        nlohmann-json-dev \
-        python3-paramiko
+        nlohmann-json-dev
 
 RUN pip3 install redis
```
Comment (andriy-kokhan) [inline Dockerfile.saithrift:18]: in base image as python3-redis

File: Dockerfile.saithrift [hunk 4]
```diff
@@ -15,43 +11,14 @@ ENV DEBIAN_FRONTEND=noninteractive
...
 
-# Add support for supervisord to handle startup dependencies
-RUN pip3 install supervisord-dependent-startup==1.4.0
-
 RUN pip3 install ctypesgen
 RUN apt-get install -y redis-server
```
Comment (andriy-kokhan) [inline Dockerfile.saithrift:21]: in base image

File: Dockerfile.saithrift [hunk 5]
```diff
@@ -133,14 +84,20 @@ RUN git clone https://github.com/sonic-net/sonic-sairedis.git \
         && cd .. \
         && dpkg -i libsaivs_1.0.0_amd64.deb \
```
Comment (andriy-kokhan) [inline Dockerfile.saithrift:85]: sonic-sairedis should be in target folder

File: Dockerfile.saithrift [hunk 6]
```diff
@@ -151,47 +108,12 @@ RUN cd /sai/gen_attr_list \
...
 # Setup supervisord
-ENV SC_TARGET_PATH=npu/${SC_PLATFORM}/${SC_ASIC}/${SC_TARGET}
-COPY ${SC_TARGET_PATH}/configs/sai.profile                 /etc/sai.d/sai.profile
-COPY ${SC_TARGET_PATH}/configs/lanemap.ini                 /usr/share/sonic/hwsku/lanemap.ini
-COPY ${SC_TARGET_PATH}/configs/port_config.ini             /usr/share/sonic/hwsku/port_config.ini
-COPY ${SC_TARGET_PATH}/configs/supervisord.conf.saithrift  /etc/supervisor/conf.d/supervisord.conf
 COPY scripts/veth-create.sh                                /usr/bin/veth-create.sh
 
 
 RUN pip3 install scapy pysubnettree pytest
```
Comment (andriy-kokhan) [inline Dockerfile.saithrift:114]: in base image

File: Dockerfile.saithrift [hunk 7]
```diff
@@ -1,100 +1,19 @@
...
-        && ./autogen.sh \
-        && dpkg-buildpackage -us -uc -b --jobs=auto \
-        && cd .. \
-        && dpkg -i libswsscommon_1.0.0_amd64.deb \
-        && dpkg -i libswsscommon-dev_1.0.0_amd64.deb \
-        && rm -rf sonic-swss-common \
-        && rm -f *swsscommon*
-
 ## Install Thrift saiserver
 ENV platform=vs
```
Comment (andriy-kokhan) [inline Dockerfile.saithrift:18]: Do we need it here?

File: Dockerfile.saithrift [hunk 8]
```diff
@@ -1,100 +1,19 @@
...
-        git \
+RUN apt-get install -y \
         nlohmann-json-dev \
-        python3-paramiko
-
-RUN pip3 install redis
-
-# Add support for supervisord to handle startup dependencies
-RUN pip3 install supervisord-dependent-startup==1.4.0
+        libtemplate-perl \
```
Comment (andriy-kokhan) [inline Dockerfile.saithrift:6]: Why do we need these packages? Are they using to build something or in runtime as well?

File: Dockerfile.saithrift [hunk 9]
```diff
@@ -110,88 +29,13 @@ RUN wget "http://archive.apache.org/dist/thrift/0.11.0/thrift-0.11.0.tar.gz" \
     --with-ruby=no --with-haskell=no --with-go=no --with-rs=no --with-haxe=no \
```
Comment (andriy-kokhan) [inline Dockerfile.saithrift:29]: Please fix indentation in L27..L30

File: Dockerfile.saithrift [hunk 10]
```diff
@@ -110,88 +29,13 @@ RUN wget "http://archive.apache.org/dist/thrift/0.11.0/thrift-0.11.0.tar.gz" \
...
-# Deploy a remote commands listener
-COPY scripts/redis-cmd-listener.py   /sai-challenger/scripts/redis-cmd-listener.py
-
-# Install SAI Challenger
-RUN pip3 install /sai-challenger/common /sai-challenger
+    && pip3 install lib/py \
+    && cd /sai \
+    && rm -rf thrift-0.11.0 thrift-0.11.0.tar.gz
 
+RUN pip3 install pysubnettree
```
Comment (andriy-kokhan) [inline Dockerfile.saithrift:36]: why do we need this?

File: Dockerfile.saithrift [hunk 11]
```diff
@@ -110,88 +29,13 @@ RUN wget "http://archive.apache.org/dist/thrift/0.11.0/thrift-0.11.0.tar.gz" \
...
 
 WORKDIR /sai-challenger
 
 CMD ["/usr/bin/supervisord"]
+
```
Comment (andriy-kokhan) [inline Dockerfile.saithrift:41]: Do not see saithrift client to be installed...

File: Dockerfile.saithrift (hunk 10 above)
Comment (taras-keryk-plv) [reply Dockerfile.saithrift]: It seems no.. Removed

File: build.sh [hunk 12]
```diff
@@ -130,11 +124,14 @@ fi
 
 # Build target Docker image
 pushd "${ASIC_PATH}/${TARGET}"
+IMG_NAME=$(echo "${ASIC_TYPE}-${TARGET}" | tr '[:upper:]' '[:lower:]')
 if [ "${IMAGE_TYPE}" = "standalone" ]; then
-    docker build -f Dockerfile -t sc-${ASIC_TYPE}-${TARGET} .
+    if [ "${SAI_INTERFACE}" = "thrift" ]; then
+        docker build -f Dockerfile.saithrift -t sc-${IMG_NAME} .
```
Comment (andriy-kokhan) [inline build.sh:130]: please have a different image name for thrift vs redis

File: npu/broadcom/BCM56850/saivs/Dockerfile.saithrift [hunk 13]
```diff
@@ -0,0 +1,90 @@
...
+
+RUN pip3 install ctypesgen
+
+WORKDIR /sai
+
+ENV SAITHRIFTV2=y
+ENV platform=vs
+
+## Install Thrift code gen
+RUN wget "http://archive.apache.org/dist/thrift/0.11.0/thrift-0.11.0.tar.gz" \
```
Comment (andriy-kokhan) [inline npu/broadcom/BCM56850/saivs/Dockerfile.saithrift:22]: Can we install this from the package? As an option, we can comment this out (do not remove for now)...

File: Dockerfile.saithrift (hunk 8 above)
Comment (taras-keryk-plv) [reply Dockerfile.saithrift]: libtemplate-perl need for gensairpc.pl in SAI repo

File: npu/broadcom/BCM56850/saivs/Dockerfile.saithrift (hunk 13 above)
Comment (taras-keryk-plv) [reply npu/broadcom/BCM56850/saivs/Dockerfile.saithrift]: Yes, we can. For correct test run,
except apt libthrift-dev libthrift-0.11.0 thrift-compiler
we need pip3 install thrift

File: build.sh (hunk 12 above)
Comment (taras-keryk-plv) [reply build.sh]: ready, please take a look

**Pull request closed without merging**